Code/GraphMol/ChemReactions/Wrap/testReactionWrapper.py

```diff
@@ -1035,6 +1034,46 @@ def testReactionInPlace(self):
     self.assertFalse(rxn.RunReactantInPlace(reactant))
     self.assertEqual(Chem.MolToSmiles(reactant), 'CCOC(C)=O')
 
+  def testGithub4651(self):
+    mol_sulfonylchloride = Chem.MolFromSmiles("Nc1c(CCCSNCC)cc(cc1)S(=O)(=O)Cl")
+    mol_amine = Chem.MolFromSmiles("Nc2cc(C)on2")
+    mol_sulfonamide = Chem.MolFromSmiles("CCNSCCCc1cc(S(=O)(=O)Nc2cc(C)on2)ccc1N")
+
+    smirks_fwd = (
+        "[S;$(S(=O)(=O)[C,c,N]):1](=O)(=O)(-[Cl])"
+         "."
+         "[N;$([N&H2&D1,N&H1&D2])"
+         # N aliphatic and not aromatic bond to carbon
+         "&$(N(-&!@[#6]))"
+         "&!$(N-C=[O,N,S]):2]"
+         ">>"
+         "[S:1](=O)(=O)-[N+0:2]"
+    )
+
+    smirks_reverse = ">>".join(smirks_fwd.split(">>")[::-1])
+
+    reaction_fwd = rdChemReactions.ReactionFromSmarts(smirks_fwd)
+    reaction_reverse = rdChemReactions.ReactionFromSmarts(smirks_reverse)
+
+
+    product = reaction_fwd.RunReactants((mol_sulfonylchloride, mol_amine))[0][0]
+    print("reaction_fwd", Chem.MolToSmiles(product))
+
+    reagent_sulfonylchloride, reagent_amine = reaction_reverse.RunReactants((mol_sulfonamide,))[0]
+
+    # trigger bug
+    print("trigger bug and mess up reaction_fwd object")
+    try:
+        product = reaction_fwd.RunReactants((reagent_sulfonylchloride, reagent_amine))[0]
+    except RuntimeError as e:
```

Comment:
if you're expecting the exceptions to be thrown here, and I guess you are, then it seems like you should be testing for the exception and not just ignoring it.

Reply:
I'll add that but for clarification the test was don't block and timeout, the exception is secondary.